src/Repository/index.js

```diff
@@ -140,41 +150,41 @@ class Repository {
         await Promise.all(createOrUpdateFilesMultiple)
     }
 
-    async createPullRequest({ title, body, branchName, defaultBranch }) {
+    async createPullRequest({ title, body, branchName }) {
         const result = await this.github.pulls.create({
             owner: this.owner,
             repo: this.repo,
             title,
             body,
             head: branchName,
-            base: defaultBranch,
+            base: this.defaultBranch,
             maintainer_can_modify: true,
         })
         return result.data.html_url
     }
 
-    async createPullRequestFromFiles({
-        title,
-        body,
-        filesByPath,
-        branchName,
-        defaultBranch,
-    }) {
-        await this.createBranch({ branchName, defaultBranch })
+    async createPullRequestFromFiles({ title, body, filesByPath, branchName }) {
+        if (this.basedBranch === this.defaultBranch)
```

Comment:
I get what your trying to go for here but it took me a while to understand.
Could we do something like
```
const branchNameExists = branchName === this.basedBranch
if (!branchNameExists) {
```